File: app/src/main/java/com/aspark/carebuddy/view/user/UserLoginActivity.java
package com.aspark.carebuddy.view.user;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.aspark.carebuddy.Contract;
import com.aspark.carebuddy.R;
import com.aspark.carebuddy.presenter.LoginPresenter;
import com.aspark.carebuddy.presenter.UserLoginPresenter;

public class UserLoginActivity extends AppCompatActivity implements Contract.View.UserLoginView {

    EditText emailEditText,passwordEditText;
    Button loginBtn;
    TextView signUpTextView,loginErrorTextView;
    String sEmail,sPassword;
    CardView loginErrorCard;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_login);

        emailEditText = findViewById(R.id.userLoginEmail);
        passwordEditText = findViewById(R.id.userLoginPassword);
        loginBtn = findViewById(R.id.userLoginBtn);
        signUpTextView = findViewById(R.id.signUpTextView);
        loginErrorCard = findViewById(R.id.loginErrorCardView);
        loginErrorTextView = findViewById(R.id.loginErrorTextView);

        hideUserLoginError();

        if (getIntent().getStringExtra("emailSent") !=null) {
            loginErrorTextView.setTextColor(Color.GRAY);
            loginErrorTextView.setText("Verification email sent successfully");
            loginErrorCard.setVisibility(View.VISIBLE);
           // showUserLoginError("Verification email sent successfully");

        }

        loginBtn.setOnClickListener(view -> {

            sEmail = emailEditText.getText().toString();
            sPassword = passwordEditText.getText().toString();

            Contract.Presenter.PresenterUserLogin presenter = new UserLoginPresenter(this);
            presenter.userLoginClickListener(this,sEmail,sPassword);

        });

        signUpTextView.setOnClickListener(view -> {
            Contract.Presenter presenter = new LoginPresenter();
            presenter.signUpBtnClickListener(this);
        });

    }

    @Override
    public void showUserLoginError(String message) {

        loginErrorTextView.setTextColor(getResources().getColor(R.color.error,getTheme()));
        loginErrorTextView.setText(message);
        loginErrorCard.setVisibility(View.VISIBLE);

    }

    @Override
    public void hideUserLoginError() {

        loginErrorCard.setVisibility(View.INVISIBLE);

    }
}

File: app/src/main/java/com/aspark/carebuddy/login/UserLoginActivity.kt
package com.aspark.carebuddy.login

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.core.widget.doAfterTextChanged
import com.aspark.carebuddy.R
import com.aspark.carebuddy.databinding.ActivityUserLoginBinding
import com.aspark.carebuddy.signup.SignUpUserActivity
import com.aspark.carebuddy.view.user.UserHomeActivity

class UserLoginActivity : AppCompatActivity() {

    private lateinit var binding: ActivityUserLoginBinding
    private val viewModel: UserLoginViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityUserLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //hideUserLoginError()

        if (intent.getStringExtra("emailSent") != null) {

            binding.tvLoginError.setTextColor(Color.GRAY)
            binding.tvLoginError.setText(R.string.email_verified_success)
            binding.cvLoginError.visibility = View.VISIBLE
            // showUserLoginError("Verification email sent successfully");
        }

        viewModel.callActivity.observe(this) {

            val intent =
                Intent(this, UserHomeActivity::class.java)
            //destroys all previous tasks
            intent.flags =
                Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(intent)
            finish()
        }

        viewModel.loginErrorMessage.observe(this){

            it?.let {
                showUserLoginError(it)
            }
        }

        binding.userLoginBtn.setOnClickListener {

           val sEmail = binding.userLoginEmail.text.toString()
           val sPassword = binding.userLoginPassword.text.toString()

            viewModel.userLoginClickListener(sEmail,sPassword)
        }

        binding.signUpTextView.setOnClickListener {
            val intent = Intent(this, SignUpUserActivity::class.java)
            startActivity(intent)
        }

        binding.userLoginEmail.doAfterTextChanged {
            hideUserLoginError()
        }
        binding.userLoginPassword.doAfterTextChanged {
            hideUserLoginError()
        }

    }

    private fun showUserLoginError(message: String?) {
        binding.tvLoginError.setTextColor(resources.getColor(R.color.error, theme))
        binding.tvLoginError.text = message
        binding.cvLoginError.visibility = View.VISIBLE
    }

     private fun hideUserLoginError() {
        binding.cvLoginError.visibility = View.INVISIBLE
    }
}